lib/src/has_next_iterator.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
```

Comment:
I would really like to avoid adding a custom iterator class here - or we should at least just wrap another iterator and implement the actual `Iterator` interface.

I would be happy to take on this part of the change if you would like though.

Reply:
It's your call. If you have a better handle on it, then by all means. This was more of a quick and dirty implementation. If it should wrap an existing `Iterator`, then by all means, feel free to address.

Reply:
Ok I went ahead and pushed up a version that just wraps another iterator 👍

Reply:
Thanks for that.

Reply:
I ended up deciding to name it `hasCurrent` and change the semantics a tiny bit as well, but basically same same :).